### ISSUE TITLE
Add **kwargs to `augment_*` methods in all BaseImageAugmentationLayer subclasses

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#487]
In subclass layers we should add **kwargs to the augment_* methods in all BaseImageAugmentationLayer subclasses -> or at least recommend users do this in https://keras.io/guides/keras_cv/custom_image_augmentations/

The reason for this is that we will be adding new arguments to base image augmentation augment_* method with future label types, including segmaps, etc.

As a first step, lets add this recommendation to the tutorial, and mention it in the "Common gotchas" section of the tutorial

**Comment by qlzh727** [comment]
I think this is all fixed now.
